### DISCUSSION TITLE
Implementing Cache Replacement Policies Gem5

### DISCUSSION THREAD

**Discussion by sahilkhan123** [gem5/gem5#1807] [Q&A]
Hi, I seek to implement a custom cache replacement policy in Gem5 and compare it to some other baseline policies (LFU, LRU, Random Replacement, etc.), and I'm stumbling upon two issues because the docs for cache replacement are very short: 

1) There seems to exist compiler flag options for the baseline policies, and you can confirm they exist by using the option to list available cache replacement policies, yet there's nothing on actually using those policies. How am I supposed to run a benchmark across these?

2) How would I go about implementing a custom cache replacement policy? Would overriding reset(), touch(), invalidate() and getVictim() in a file like se.py be sufficient?

Any help on any of these questions would be great. Thanks!

**Comment by ZhanyueZ** [comment]
Hi Did you figure out how?

**Comment by sahilkhan123** [comment]
Hi,

Yes. Thanks.
